src/app/shared/models/domain/item-edit-permission.ts

```diff
@@ -0,0 +1,56 @@
+import { Pipe, PipeTransform } from '@angular/core';
+import * as D from 'io-ts/Decoder';
+
+// short versions, not for export
+const
+  NONE = 'none',
+  CHILDREN = 'children',
+  ALL = 'all',
+  ALL_WITH_GRANT = 'all_with_grant';
+
+export const
+  ITEMEDITPERM_NONE = NONE,
+  ITEMEDITPERM_CHILDREN = CHILDREN,
+  ITEMEDITPERM_ALL = ALL,
+  ITEMEDITPERM_ALL_WITH_GRANT = ALL_WITH_GRANT,
+  ITEMEDITPERM_MAX = ALL_WITH_GRANT;
```

Comment:
I suppose you have considered using enums and dropped the idea, for my information, why?

In my opinion uppercase variable names like these are not very readable.
As I understand it, the idea is to follow the constant naming convention, that would give `ITEM_EDIT_PERM_*` instead which makes very long names now and tons of imports in some other files
I would rather picture exporting an object `itemEditPermissions` - or `itemEditPerm`, `editPermissions`, `editPerm`, whatever - containing the values

Reply:
FYI, you can also use the `export as` notation:
```ts
export {
   NONE as ITEMEDITPERM_NONE,
 }
```
I'm not sure it makes much a difference though :thinking: